extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -334,7 +335,11 @@ async function runRemoteQueriesApiRequest(
     void showAndLogInformationMessage(popupMessage, { fullMessage: logMessage });
     return response.data.workflow_run_id;
   } catch (error) {
-    void showAndLogErrorMessage(getErrorMessage(error));
+    if ((error as Response).status === 404) {
```

Comment:
Hmm what made you use `Response` from `node-fetch`? I think that response type is probably `RequestError` (https://octokit.github.io/types.ts/modules/RequestError.html) but I notice we don't have `@octokit/types` in our package.json. They both have `status` so TypeScript is happy.

I suggest we do the following:
- Bring in `@octokit/types` as a dev dependency and try to use `RequestError` from there.
- If for some reason that causes you headache, just go with `any` instead of `Response`:

```typescript
catch (error: any) {
  if (error.status === 404) {
   ...
}
```

Reply:
Thanks! Had a look at this with @shati-patel and we realized we're already using `OctokitResponse<QueriesResponse, number>)` when we make the API call in the same `try ... catch()` block. Using that type builds successfully so we didn't need to install `@octokit/types` as a dev dependency in the end. 

We did try to add `@octokit/types` initially but at the moment this is causing a Typescript error in another plugin:

```
/Users/elenatanasoiu/projects/vscode-codeql/extensions/ql-vscode/node_modules/@octokit/rest/node_modules/@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types.d.ts(425,60): error TS2339: Property 'POST 
```

There's an issue opened 20 hours ago on this, but it's yet unresolved: https://github.com/octokit/plugin-rest-endpoint-methods.js/issues/488

Reply:
🤔 we do use `OctokitResponse<QueriesResponse, number>)` but as the return type of `octokit.request()`. Here, we're trying to access the type of the error.

It's a shame that the typings are broken - thanks for checking though!

This is pretty minor but I think it could cause confusion. I'd rather we use `any` than the wrong type. What do you think?

Reply:
Fine by me! 😉 Have changed.